OSPSuite.Presentation.Importer/Presenters/SourceFilePresenter.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.IO;
+using OSPSuite.Presentation.Importer.Views;
+using OSPSuite.Presentation.Presenters;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Presentation.Importer.Presenters
+{
+   public class SourceFilePresenter : AbstractPresenter<ISourceFileControl, ISourceFilePresenter>, ISourceFilePresenter
+   {
+      private readonly IDialogCreator _dialogCreator;
+      
+      public string Title  { get; set;} 
+      public string Filter { get; set; }
+      public string DirectoryKey { get; set; }
+
+
+      public SourceFilePresenter(IDialogCreator dialogCreator, ISourceFileControl view) : base(view)
+      {
+         _dialogCreator = dialogCreator;
+         Title = string.Empty;
+         Filter = string.Empty;
+         DirectoryKey = string.Empty;
+      }
+
+      public void OpenFileDialog( string initFileName)
+      {
+         string initDirectoryName = null;
+         if (!initFileName.IsNullOrEmpty()) 
+            initDirectoryName = Path.GetDirectoryName(initFileName);
+         OnSourceFileChanged.Invoke(this, new SourceFileChangedEventArgs() { FileName = _dialogCreator.AskForFileToOpen(Title, Filter, DirectoryKey, null, initDirectoryName) });
```

Comment:
why the invoke?

Reply:
For better thread safety reasons. We are also using it in quite a few other places. It is again a point we could discuss in a meeting.

Reply:
I dont't think it can be null if it is instnatiated as an empty delegate. (this is threadsafe by construction)
Or what do you mean ?